mlrun/platforms/iguazio.py

```diff
@@ -582,3 +582,48 @@ def sanitize_username(username: str):
     So simply replace it with dash
     """
     return username.replace("_", "-")
+
+
+def min_iguazio_versions(*versions):
+    def decorator(function):
+        def wrapper(*args, **kwargs):
+            if validate_iguazio_version_compatibility(*versions):
+                return function(*args, **kwargs)
+
+            message = (
+                f"{function.__name__} is supported since Iguazio {' or '.join(versions)}, currently using "
+                f"Iguazio {mlconf.nuclio_version}."
+            )
+            raise mlrun.errors.MLRunIncompatibleVersionError(message)
+
+        return wrapper
+
+    return decorator
+
+
+def validate_iguazio_version_compatibility(*min_versions):
```

Comment:
I feel like this is code duplication maybe we can merge it with `validate_nuclio_version_compatibility`